packages/devtools_app/lib/src/timeline/flutter/timeline_processor.dart

```diff
@@ -654,11 +117,21 @@ class FullTimelineProcessor extends TimelineProcessor {
 //        .map((e) => TraceEventWrapper(
 //            TraceEvent(e), DateTime.now().microsecondsSinceEpoch))
 //        .toList();
-    final _traceEvents = (traceEvents
-        // Throw out timeline events that do not have a timestamp
-        // (e.g. thread_name events).
-        .where((event) => event.event.timestampMicros != null)
-        .toList())
+    final _traceEvents = (traceEvents.where((event) {
+      // Throw out 'MessageLoop::FlushTasks' events. A single
+      // 'MessageLoop::FlushTasks' event can parent multiple Flutter frame event
+      // sequences and complicates the frame detection logic.
+      final isMessageLoopFlushTasks =
+          event.event.name.contains('MessageLoop::FlushTasks');
```

Comment:
Maybe move all the magic event name constants e.g (`'MessageLoop::FlushTasks'`) we match to be top level fields in this library or a helper library with a prefix. That way you can audit at a glance all the special cases we are handling differently.